src/rt/aaA.d

```diff
@@ -684,6 +684,10 @@ extern (C) pure nothrow @nogc
                 break;
         }
     }
+    size_t _aaRangeLen(Range r)
+    {
+        return r.impl is null ? 0 : r.impl.length;
```

Comment:
No. You need a new field. Idx is just the bucket and doesn't have any relation to the number of elements in the range

Sent from my iPhone

> On Apr 3, 2016, at 7:55 PM, yshui <notifications@github.com> wrote:
> 
> In src/rt/aaA.d:
> 
> > @@ -684,6 +684,10 @@ extern (C) pure nothrow @nogc
> >                  break;
> >          }
> >      }
> > +    size_t _aaRangeLen(Range r)
> > +    {
> > +        return r.impl is null ? 0 : r.impl.length;
> @schveiguy So used - deleted - idx?
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
>